music/demos/melody_controls.ts

```diff
@@ -166,6 +171,19 @@ async function runMelMulti() {
   writeTimer('mel-multi-sample-time', start);
   writeNoteSeqs('mel-multi-seqs', [rhythmSeq, shapeSeq, melodySeqs[0]]);
 
+  const melodySeqsIonian = await mvaeMelKey.sample(
+      1, null, ['C'], undefined, undefined, controlTensor, 0);
+  const zKey =
+      await mvaeMelKey.encode(melodySeqsIonian, ['C'], controlTensor, 0);
+  const melodySeqsLydian = await mvaeMelKey.decode(
+      zKey, null, ['C'], undefined, undefined, controlTensor, 7);
+  const melodySeqsMixolydian = await mvaeMelKey.decode(
```

Comment:
Just to straighten out the conflation of chord / key... Is the thought process that the combination of 7 (which I thought I read to mean GMaj) _plus_ the CMaj expected to push the model to generate notes in MixoLydian? There's not an explicit mode conditioning signal, correct?

I'm thinking of the case where you'd want to retain the key but change the mode... would you do that by holding the `key` number the same while adjusting the chord progression conditioning signal?

Reply:
By retain the key but change the mode, do you mean e.g. playing in C major but switch the underlying chord from C (Ionian) to G (Mixolydian), or do you mean playing in a mode rooted at C but change from C Ionian to C Dorian etc.?  If the first, then yes you would retain key but change the chord.

Reply:
I'm more thinking about the latter... From a UX perspective, I'm imagining that it might be fun for the user to switch modes "on-the-fly", and in my mind that operation keeps the root note the same (switch from C Ionian to C Aeolian). It's pretty fun to swap relative-encoded melodies between modes.

In this case do you just do something like this? 

```typescript
decode(zKey, null, ['C'], undefined, undefined, controlTensor, 0) // C Ionian
decode(zKey, null, ['Cmin'], undefined, undefined, controlTensor, 0) // C Aeolian

or is it

decode(zKey, null, ['C'], undefined, undefined, controlTensor, 0) // C Ionian
decode(zKey, null, ['Cmin'], undefined, undefined, controlTensor, 9) // C Aeolian
```

Reply:
It would be neither of those :)  You'd pass 0 as the key the first time and 3 (key of Eb) the second time.

Reply:
Sorry, maybe we're not on the same page here but I'm confused. How would you express switching the conditioning of a melody from C Major to C minor?

Reply:
Nevermind, I see it now with the comment for `MusicVAEControlArgs`... there's no other modal conditioning, though, correct?

Reply:
Correct, there's no mode, just chord and key.